src/string_id.h

```diff
@@ -127,6 +127,15 @@ class string_id
          * Returns the actual object this id refers to. May show a debug message if the id is invalid.
          */
         const T &obj() const;
+
+        const T &operator*() const {
+            return obj();
+
+        }
+        const T *operator->() const {
+            return &obj();
+        }
```

Comment:
> It's a lot less ugly than foo.obj().quux.obj()
> Making them use less symbols will not make them less ugly

`string_id<foo>::obj` is an extra symbol we don't need. `->` and `.` are for many people much easier to mentally parse. 

> It obfuscates the type system underneath

If you don't know the underlying type you have already lost. The usual culprits are excessive use of `auto`, deep nesting or excessively long functions. We have plentiful examples of all of those in the codebase:
```
const auto quux = item::get_something()`
...
quux->action(); // what am I?
quux.obj().action(); // even less clear
```

> I can merge it in the current state if you're willing to take all the blame

This PR has been open a while now and has had a good chance for review including by the original author of the class. Responsibility for the code is naturally with myself. The same system has worked well for `item_location` which after extensive usage for sometime has not resulted in any bugs from the indirection operators nor raised any prior objections.